manual/src/refman/patterns.etex

```diff
@@ -199,6 +308,12 @@ Matching a value with a @pattern-matching@ where some patterns
 contain @"lazy"@ sub-patterns may imply forcing parts of the value,
 even when the pattern selected in the end has no @"lazy"@ sub-pattern.
 
+\begin{caml_example}{toplevel}
+let lazy_option_map f = function
+  | lazy (Some x) -> Some (Lazy.force f x)
+  | _ -> None;;
+\end{caml_example}
```

Comment:
Actually this implementation is "wrong", it forces the thunk on application where there is no need for mapping.

You want to write:
```ocaml
let lazy_option_map f lo =
  lazy (match lo with
    | lazy (Some x) -> let (lazy y) = f x in Some y
    | lazy None -> None)
```

Note: I think `Lazy.force f x` was meant to be `Lazy.force (f x)`, as the current one makes the type not look like a map.

I think that there are simpler examples for functions that force their lazy inputs, instead of trying to preserve laziness.

```ocaml
let force_opt : int lazy option -> int = function
| Some (lazy n) -> n
| None -> 0
```

Reply:
The example is taken (modified only slightly) from the Lazy modules' documentation:

https://caml.inria.fr/pub/docs/manual-ocaml/libref/Lazy.html

Can you suggest an example suitable both for the Patterns chapter, and to fix the Laxy documentation (I've never used Lazy)?